cdap-app-fabric/src/main/java/co/cask/cdap/metadata/LineageAdmin.java

```diff
@@ -158,7 +159,8 @@ public Lineage computeLineage(final StreamId sourceStream, long startMillis, lon
   public Set<MetadataRecord> getMetadataForRun(ProgramRunId run) throws NotFoundException {
     entityExistenceVerifier.ensureExists(run);
 
-    Set<NamespacedEntityId> runEntities = new HashSet<>(lineageStoreReader.getEntitiesForRun(run));
+    Set<MetadataEntity> runEntities = new HashSet<>();
+    lineageStoreReader.getEntitiesForRun(run).forEach(x -> runEntities.add(x.toMetadataEntity()));
```

Comment:
Using `forEach` to add stuff is not recommended. You should use collector in this case.

```
Set<MetadataEntity> runEntities = lineageStoreReader.getEntitiesForRun(run).stream()
  .map(NamespacedEntityId::toMetadataEntity)
  .collect(Collectors.toSet());
```